lib/kaitai/struct/struct.rb

```diff
@@ -531,6 +531,22 @@ def to_signed(x, mask)
   def self.format_hex(bytes)
     bytes.unpack('H*')[0].gsub(/(..)/, '\1 ').chop
   end
+
+  ###
+  # Guess about given arg being a byte-array most likely.
+  # <p>
+  # There's no way to know for sure in Ruby, because the type {@code String} is used in both cases,
+  # but especially in the context of KS things are differently likely. This especially means that an
```

Comment:
Um, what do you mean by that

> but especially in the context of KS **things are differently likely**.

...? Did you mean it like that _things_ have _different likelihood_ in the context of KS than in the rest of the world?

In any case, I consider this a quite clumsy formulation. And I don't think that the usage of the `ASCII-8BIT` encoding for character strings in the _rest of the world_ is somewhere high. If someone uses it with this intent, most likely they meant `US-ASCII` instead and they simply don't know [the difference](https://idiosyncratic-ruby.com/56-us-ascii-8bit.html):

> So, **US-ASCII** is aliased to **ASCII**, but then what is **ASCII-8BIT** for? [Encodings' RDoc](https://ruby-doc.org/core/Encoding.html) has some help:
>
> [...]
>
> So basically, it is not a real encoding, but represents an arbitrary stream of bytes (bytes with a value between 0 and 255). It is used for raw byte stream or if you want to make clear that you do not know about a string's encoding!

To me, this looks like an exact definition of a byte _string_ (or array). I don't think that anyone understanding what `ASCII-8BIT` means in Ruby would expect that it will be treated and printed as a string. Besides, if you try to normally print a byte string (without anything like `.inspect` that sometimes does some [weird escaping](https://github.com/kaitai-io/kaitai_struct_ruby_runtime/pull/7#issuecomment-793155345)), you'll see this:

```ruby
puts [0x8b, 0x99, 0xab, 0xe9].pack('C*').encoding # => ASCII-8BIT
puts [0x8b, 0x99, 0xab, 0xe9].pack('C*') # => ����
```

Are four diamonds better than a hex dump, i.e. is it better to treat it is a "human-readable string" (that turns out to be non-readable in this case) or a byte string?

For these reasons, I would shorten this description quite a bit. Just the first sentence from https://ruby-doc.org/core/Encoding.html is enough to describe the method in my opinion:

> Encoding::ASCII_8BIT is a special encoding that is usually used for a byte string, not a character string.

---

> but especially in the context of KS things are differently likely.

> is considered to be VERY unlikely for human readable texts.

Phrases "differently likely" or "VERY unlikely" here misuse the term "un/likely" (as a degree of probability) anyway: [_probability_](https://en.wikipedia.org/wiki/Probability) only makes sense for events that can be considered random, which does not really apply for programming habits. Programmers (if they're not trolling) use each feature of a language for a reason. They usually don't roll a dice and say: "if I roll 6, I'll use `ASCII-8BIT` for my ASCII string in my Ruby program, otherwise I use `US-ASCII` _(like every normal person)_". Sure, they all solve different problems with different priorities, but it can be objectively observed that some approaches are more suitable than others. And trust me, `ASCII-8BIT` is inappropriate for **character** strings. Prove me wrong.